src/worker.jl

```diff
@@ -417,24 +433,34 @@ end
 ##############################       HANDLER FUNCTIONS        ##############################
 
 """
-    get_data(worker::Worker; keys::Array=[], who::String="")
+    get_data(worker::Worker; keys::Array=String[], who::String="") -> Dict
 
 Send the results of `keys` back over the stream they were requested on.
+
+# Returns
+- `Dict{Vector{UInt8}, Vector{UInt8}}`: dictionary mapping keys to their serialized data for
+    communication
 """
-function get_data(worker::Worker; keys::Array=[], who::String="")
+function get_data(worker::Worker; keys::Array=String[], who::String="")
+    keys = Vector{String}(keys)
     debug(logger, "\"get_data\": ($keys: \"$who\")")
-    return Dict(
-        to_key(k) =>
+    return Dict{Vector{UInt8}, Vector{UInt8}}(
+        Vector{UInt8}(k) =>
         to_serialize(worker.data[k]) for k in filter(k -> haskey(worker.data, k), keys)
     )
 end
 
 """
-    gather(worker::Worker; who_has::Dict=Dict())
+    gather(worker::Worker; who_has::Dict=Dict{String, Vector{String}}())
 
 Gather the results for various keys.
 """
-function gather(worker::Worker; who_has::Dict=Dict())
+function gather(
+    worker::Worker;
+    who_has::Dict=Dict{String, Vector{String}}()
+)::Dict
+
+    who_has = Dict{String, Vector{String}}(who_has)
```

Comment:
Sometimes (in either julia 0.5 or 0.6 I forget which right now) the input to the function is just a Dict of type `{Any, Any}` because of how Msgpack interprets the message recieved from the scheduler. So it's not until we are inside this function that it can be cast to its proper type or just worked with as is.